### PR TITLE
uzlib(trezor): fix compilation with clang

### PR DESCRIPTION
**Pull request by mmilata**

### CODE REVIEW

File: lib/uzlib/tinflate.c [hunk 1]
```diff
@@ -35,6 +35,12 @@
 #include <assert.h>
 #include "tinf.h"
 
+#if defined(__GNUC__) && (__GNUC__ >= 11)
```
Comment (prusnak) [inline lib/uzlib/tinflate.c]: Why `__GNUC__ >= 11`?

It seems that optimize was introduced in GCC 4.4.0: https://gcc.gnu.org/onlinedocs/gcc-4.4.0/gcc/Function-Attributes.html#index-g_t_0040code_007boptimize_007d-function-attribute-2243

**Comment by prusnak** [comment]
Btw, isn't it worth using -O3 for the whole `lib/uzlib/tinflate.c` file?

```
#pragma GCC optimize("O3")
```

That should work for all compilers (although this will be ignored by all other compilers than GCC).

File: lib/uzlib/tinflate.c (hunk 1 above)
Comment (mmilata) [reply lib/uzlib/tinflate.c]: Without the version comparison compilation fails with the gcc version that's installed on the darwin CI runner. Will change it to `>= 5`.

**Comment by mmilata** [comment]
I'm just trying to fix the build. @TychoVrahe have you tried this?^

**Comment by TychoVrahe** [comment]
yeah i tried optimizing the whole file - it increases the binary size by some 2kB, while most of the speed gains are from optimizing this one function, so it didn't seem like a good tradeoff.

We are in a bit better spot with flash space right now so maybe we can reconsider, but the conditional seems ok to me too.

**Comment by prusnak** [comment]
Ok, let's use the proposed conditional then.

**Review by matejcik — APPROVED** [review]
utACK

**Review by prusnak — APPROVED** [review]
utACK

**Pull request merged**